src/Compilers/Core/Portable/Text/SourceText.cs

```diff
@@ -102,21 +102,56 @@ public static SourceText From(string text, Encoding? encoding = null, SourceHash
             return new StringText(text, encoding, checksumAlgorithm: checksumAlgorithm);
         }
 
+        /// <summary>
+        /// Constructs a <see cref="SourceText"/> from text in a <see cref="StringBuilder"/>.
+        /// </summary>
+        /// <param name="stringBuilder"><see cref="StringBuilder"/> that contains the text.</param>
+        /// <param name="encoding">
+        /// Encoding of the file that the text was read from or is going to be saved to.
+        /// <see langword="null"/> if the encoding is unspecified.
+        /// If the encoding is not specified, the resulting <see cref="SourceText"/> isn't debuggable.
+        /// If an encoding-less <see cref="SourceText"/> is written to a file, <see cref="Encoding.UTF8"/> shall be used as a default.
+        /// </param>
+        /// <param name="checksumAlgorithm">
+        /// Hash algorithm to use to calculate checksum of the text that's saved to PDB.
+        /// </param>
+        /// <exception cref="ArgumentNullException"><paramref name="stringBuilder"/> is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentException"><paramref name="checksumAlgorithm"/> is not supported.</exception>
+#pragma warning disable RS0026 // Do not add multiple public overloads with optional parameters
```

Comment:
Is this OK? I know there shouldn't be multiple overloads with optional parameters, but it seems OK if they differ in the type of first parameter, which is non-optional. Overloads like this already exists here - some take `string`, some take `TextReader` and some take `Stream`, with the other parameters optional and there isn't any warning anywhere.